core/types.ml

```diff
@@ -1774,15 +1887,19 @@ struct
     [(var, spec)]
 
   let free_bound_tycon_vars ~include_aliases bound_vars tycon_spec =
+    let split_vars tyvars =
+      let bound_vars, vars =
+        List.fold_left
+          (fun (bound_vars, vars) tyvar ->
+             let var, spec = varspec_of_tyvar tyvar in
+               (TypeVarSet.add var bound_vars, (var, spec)::vars)) (bound_vars, []) tyvars in
+        (bound_vars, List.rev vars) in
+
     match tycon_spec with
       | `Alias (tyvars, body) ->
-          let bound_vars, vars =
-            List.fold_left
-              (fun (bound_vars, vars) tyvar ->
-                 let var, spec = varspec_of_tyvar tyvar in
-                   (TypeVarSet.add var bound_vars, (var, spec)::vars)) (bound_vars, []) tyvars
-          in
-            (List.rev vars) @ (free_bound_type_vars ~include_aliases bound_vars body)
+          let (bound_vars, vars) = split_vars tyvars in
+          vars @ (free_bound_type_vars ~include_aliases bound_vars body)
+      | `Mutual (tyvars, _) -> snd @@ split_vars tyvars
```

Comment:
Am I the only one who dislikes `@@`?

Reply:
`@@` is rubbish, yes. `<|` is a synonym. Would that be OK? Alternatively I could just be explicit and bracket it.

Reply:
My personal preference is to use brackets, but I don't really know what is more common in OCaml.

Reply:
Yes, just use parentheses.